connectors/sources/onedrive.py

```diff
@@ -256,18 +330,29 @@ async def list_users(self):
             for user_detail in response:
                 yield user_detail
 
-    async def get_owned_files(self, user_id):
+    async def get_owned_files(self, user_id, skipped_extensions=None, pattern=""):
         async for response in self.paginated_api_call(url_name=DELTA, user_id=user_id):
             for file in response:
                 if file.get("name", "") != "root":
-                    yield file
+                    parent_path = file.get("parentReference", {}).get("path")
+                    is_match = glob.globmatch(parent_path, pattern, flags=glob.GLOBSTAR)
+                    has_skipped_extension = os.path.splitext(file["name"])[-1] in (
+                        skipped_extensions or []
+                    )
+                    if (skipped_extensions and has_skipped_extension) or (
```

Comment:
Not necessary to check `skipped_extensions` here.

Reply:
Good catch.